internal/resources/resource_host.go

```diff
@@ -431,6 +454,37 @@ func resourceMetalHostCreate(d *schema.ResourceData, meta interface{}) (err erro
 	if err != nil {
 		return err
 	}
+
+	// host create is asynchronous in Metal svc. Wait until host state is Ready.
+	createStateConf := &resource.StateChangeConf{
+		Pending: []string{
+			hStateNew,
+			hStateImagingPrep,
+			hStateImaging,
+			hStateConnecting,
+			hStateBooting,
+		},
+		Target: []string{
+			hStateReady,
+		},
+		Refresh: func() (interface{}, string, error) {
+			host, _, err := p.Client.HostsApi.GetByID(ctx, h.ID)
+			if err != nil {
+				return nil, "", fmt.Errorf("get host %v", h.ID)
+			}
+
+			return host, string(host.State), nil
+		},
+		Timeout:    d.Timeout(schema.TimeoutCreate),
+		Delay:      thirty * time.Second,
+		MinTimeout: ten * time.Second,
+	}
+
+	_, err = createStateConf.WaitForStateContext(ctx)
+	if err != nil {
+		return fmt.Errorf("waiting for host instance (%s) to be created: %s", d.Id(), err)
+	}
+
```

Comment:
I think the d.SetId needs to come as soon as Metal has created an ID otherwise terraform will never know about it if we error out earlier

Reply:
hmm, I thought the same, but in testing this worked;  I induced a timeout, which failed tf.  I then was able delete the resource.  Will test again.

Reply:
Were you able to delete it via Terraform?

I agree with mark. Since it is created, setting ID would let perform further operations e.g delete via terraform.

Reply:
yes, already fixed in latest patch